### PR TITLE
feat: add ed25519 support to JWK (public keys)

### PR DESCRIPTION
**Pull request by edudobay**
This PR builds on https://github.com/firebase/php-jwt/pull/399 to support parsing JWK ed25519 keys.

* The base branch is temporarily set to #399 — I see this PR as a complement to that one. Please let me know if this should be done in any other way.

Reference documentation: [RFC8037](https://datatracker.ietf.org/doc/html/rfc8037)

### CODE REVIEW

File: src/JWK.php [hunk 1]
```diff
@@ -137,8 +143,28 @@ public static function parseKey(array $jwk): ?Key
...
+
+                if (!isset(self::OKP_CURVES[$jwk['crv']])) {
+                    throw new DomainException('Unrecognised or unsupported OKP curve');
+                }
+
+                if (empty($jwk['x'])) {
+                    throw new UnexpectedValueException('x not set');
+                }
+
+                $publicKey = \base64_encode(JWT::urlsafeB64Decode($jwk['x']));
```
Comment (edudobay) [inline src/JWK.php:164]: This is not very pretty — should I create a method to convert from URL-safe to "standard" base64? Should that method be placed here in the `JWK` class or in `JWT`?

Comment (edudobay) [reply src/JWK.php]: Well, I did exactly that, in the `JWT` class.

**Comment by bshaffer** [comment]
This is great! Thank you for submitting this! I'll take a look. I think for my base PR and this PR also, I'd like more robust testing cases. I am not sure how solid the code I wrote is... for instance, I was able to find cases where verification failed and I'm not sure why.

If you're up to it, and can review #399, that would help a lot!

**Comment by edudobay** [comment]
Hi @bshaffer! Sorry I've taken so long to reply. Is there anything I can do to make this mergeable? I'm not sure why this PR was closed. If you'd like any further help let me know.

**Comment by bshaffer** [comment]
Hi @edudobay 

This PR was closed automatically because I merged the `add-ec-jwk-support` branch which you submitted the PR to. If you open a new PR to the `main` branch, we can get that reviewed and merged!

**Comment by bshaffer** [comment]
We also had a separate PR which adds `ed25519` as well which has already been merged: https://github.com/firebase/php-jwt/pull/343. Please let me know if you see any way to improve this implementation!

**Pull request closed without merging**